src/js/data-builder.js

```diff
@@ -199,14 +208,12 @@ const buildObjectNodeRelationship = (renderedNodes, filteredJson) => {
   });
 
   //Return nodes that are related to provided filtered json
-  return renderedNodes.filter((node) => relatedData.some((item) => item.name === node.name));
+  return renderedNodes.filter((node) => relatedData.some((item) => item.id === node.id));
 };
 
 const filterObjectsOnNodeClick = (json, nodes = []) => {
```

Comment:
Please comment your function definitions

Reply:
Function definition is self explanatory.